modules/swagger-jaxrs/src/main/java/io/swagger/jaxrs/Reader.java

```diff
@@ -823,7 +817,22 @@ private Operation parseMethod(Class<?> cls, Method method, List<Parameter> globa
         } else if (responseType == null) {
             // pick out response from method declaration
             LOGGER.debug("picking up response class from method " + method);
+
             responseType = method.getGenericReturnType();
+
+            // Use classmate's resolvedMethod if available for finding out return type
+            // (this way generic return parameter's actual type will be picked up as well)
+            if (method.getGenericReturnType() != null && method.getGenericReturnType() instanceof TypeVariable
+                    && resolvedMethod != null && resolvedMethod.getReturnType() != null
+                    && resolvedMethod.getReturnType() instanceof ResolvedObjectType) {
+                responseType = ((ResolvedObjectType) resolvedMethod.getReturnType()).getErasedType();
+            } else if (method.getGenericReturnType() != null && method.getGenericReturnType() instanceof GenericArrayType
+                    && resolvedMethod != null && resolvedMethod.getReturnType() != null
+                    && resolvedMethod.getReturnType() instanceof ResolvedArrayType) {
+                responseType = ((ResolvedArrayType) resolvedMethod.getReturnType()).getErasedType();
+            }
+
+            LOGGER.debug("response class of method " + method + " is " + responseType);
```

Comment:
Could you use`LOGGER.debug("response class of method {} is {}", method, responseType`? https://www.slf4j.org/faq.html#logging_performance